src/ui/public/agg_types/buckets/terms_other_bucket_helper.js

```diff
@@ -0,0 +1,143 @@
+  if (aggs[aggWithOtherBucket.id]) return aggs[aggWithOtherBucket.id];
+  return getAggConfig(_.values(aggs)[0].aggs, aggWithOtherBucket);
+};
+
+const getAggResultBuckets = (aggsConfig, response, aggWithOtherBucket, key) => {
+  const keyParts = key.split('-');
+  let responseAgg = response;
+  for (const i in keyParts) {
+    if (keyParts[i]) {
+      const agg = _.values(responseAgg)[0];
+      const aggKey = _.keys(responseAgg)[0];
+      const aggConfig = _.find(aggsConfig, agg => agg.id === aggKey);
+      const bucket = _.find(agg.buckets, (bucket, bucketObjKey) => {
+        const bucketKey = aggConfig.getKey(bucket, Number.isInteger(bucketObjKey) ? null : bucketObjKey).toString();
+        return bucketKey === keyParts[i];
+      });
+      if (bucket) {
+        responseAgg = bucket;
+      }
+    }
+  }
+  if (responseAgg[aggWithOtherBucket.id]) return responseAgg[aggWithOtherBucket.id].buckets;
+  return [];
+};
+
+const getAggConfigResult = (responseAggs, aggId, bucketKey) => {
+  let resultBuckets = [];
+  if (responseAggs[aggId]) {
+    const matchingBucket = responseAggs[aggId].buckets.find(bucket => bucket.key === bucketKey);
+    if (matchingBucket) resultBuckets.push(matchingBucket);
+    return resultBuckets;
+  }
+  _.each(responseAggs, agg => {
+    resultBuckets = [
+      ...resultBuckets,
+      ...getAggConfigResult(agg.aggs, aggId, bucketKey)
+    ];
+  });
+
+  return resultBuckets;
+};
+
+
+export const OtherBucketHelperProvider = (Private) => {
+  const AggConfig = Private(VisAggConfigProvider);
+
+  const buildOtherBucketAgg = (aggsConfig, aggs, aggWithOtherBucket, response) => {
+    const bucketAggs = aggsConfig.filter(agg => agg.type.type === 'buckets');
+    const index = bucketAggs.findIndex(agg => agg.id === aggWithOtherBucket.id);
+
+    // create filters aggregation
+    const filterAgg = new AggConfig(aggsConfig[index].vis, {
+      type: 'filters',
+      id: 'other',
+      schema: {
+        group: 'buckets'
+      }
+    });
+
+    // nest all the child aggregations of aggWithOtherBucket
+    const filterAggDsl = filterAgg.toDsl();
+    const resultAgg = {
+      aggs: getAggConfig(aggs, aggWithOtherBucket).aggs,
+      filters: filterAggDsl
+    };
+
+    // create filters for all parent aggregation buckets
+    const walkBucketTree = (aggIndex, aggs, aggId, filters, key) => {
+      const agg = aggs[aggId];
+      const newAggIndex = aggIndex + 1;
+      const newAgg = bucketAggs[newAggIndex];
+      const currentAgg = bucketAggs[aggIndex];
+      if (aggIndex < index) {
+        _.each(agg.buckets, (bucket, bucketObjKey) => {
+          const bucketKey = currentAgg.getKey(bucket, Number.isInteger(bucketObjKey) ? null : bucketObjKey);
+          const filter = _.cloneDeep(bucket.filter) || currentAgg.createFilter(bucketKey);
+          delete filter.meta;
+          const migratedFilter = migrateFilter(filter.query || filter);
+          const newFilters = [...filters, migratedFilter];
```

Comment:
You might want to use the [buildQueryFromFilters](https://github.com/elastic/kibana/blob/3d01a544d2b8bc9005a7ca808ce1a39919e18fb8/src/ui/public/courier/data_source/build_query/from_filters.js#L42) helper here. It'll migrate and clean (remove meta prop) the filters exactly how we do it in SearchSource and return a valid bool query body that you can use.